src/components/Filters/Filters.stories.tsx

```diff
@@ -12,139 +14,210 @@ export default {
     badges: [BADGE.BETA],
   },
   args: {
-    checkboxFields: [
-      {
-        legend: 'Filters Segment 1',
-        checkboxes: [
-          { label: 'Filters label 1', value: 'Filters label 1' },
-          { label: 'Filters label 2', value: 'Filters label 2' },
-          { label: 'Filters label 3', value: 'Filters label 3' },
-        ],
-      },
-    ],
-    onClose: (checkedValues) => {
-      console.log(checkedValues);
-    },
+    buttonText: 'Filters',
+    buttonStatusVariant: 'neutral',
+    children: (
+      <Filters.FiltersCheckboxField legend="Filters Segment 1">
+        <Checkbox
+          label="Filters label 1"
+          onChange={
+            () => {} /* eslint-disable-line @typescript-eslint/no-empty-function */
+          }
+        />
+        <Checkbox
+          label="Filters label 2"
+          onChange={
+            () => {} /* eslint-disable-line @typescript-eslint/no-empty-function */
+          }
+        />
+        <Checkbox
+          label="Filters label 3"
+          onChange={
+            () => {} /* eslint-disable-line @typescript-eslint/no-empty-function */
+          }
+        />
+      </Filters.FiltersCheckboxField>
+    ),
   },
+  decorators: [
+    (Story) => (
+      <div style={{ margin: '0.25rem', height: '100vh' }}>
+        <Story />
+      </div>
+    ),
+  ],
 } as Meta<Args>;
 
 type Args = React.ComponentProps<typeof Filters>;
 
-export const DefaultInteractive: StoryObj<Args> = {};
-
-const overflowCheckboxFields = [
-  {
-    legend: 'Filters Segment 1',
-    checkboxes: [
-      {
-        label:
-          'Filters long label 1 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod',
-        value: 'segment1label1',
-      },
-      { label: 'Filters label 2', value: 'segment1label2' },
-      { label: 'Filters label 3', value: 'segment1label3' },
-    ],
-  },
-  {
-    legend: 'Filters Segment 2',
-    checkboxes: [
-      { label: 'Filters label 1', value: 'segment2label1' },
-      {
-        label:
-          'Filters long label 2 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod',
-        value: 'segment2label2',
-      },
-      { label: 'Filters label 3', value: 'segment2label3' },
-    ],
+export const Default: StoryObj<Args> = {
+  play: async ({ canvasElement }) => {
+    // We want to test visual regression for the drawer as well as the button, but don't want the drawer open initally outside Chromatic
+    if (isChromatic()) {
+      const canvas = within(canvasElement);
+      const filtersTrigger = await canvas.findByRole('button');
+      filtersTrigger.click();
+    }
```

Comment:
Although repeated code in the other story, the play function does not execute when placed in the `export default` block